### DISCUSSION TITLE
How to filter search results

### DISCUSSION THREAD

**Discussion by earthwytch** [statamic-rad-pack/meilisearch#41] [Help] [answered]
### Description

Is there a way to use filters in a search? Typical use is to have a drop-down list to filter the results. Something like:
```
<form action="/search/results">
    <input type="search" name="q" placeholder="Search">
    <select class="pr-8 focus:outline-white" type="text" name="age" id="age">
        <option value="Any">Any</option>
         <option value="Adults">Adults</option>
         <option value="Children and young people">Children and young people</option>
    </select>
    <button type="submit">Go find it!</button>
</form>
```

In the README I can see `'filterableAttributes' => ['age'],` and I can see there is an [option to filter in the Meilisearch documentation](https://www.meilisearch.com/docs/learn/filtering_and_sorting/filter_search_results#use-filter-when-searching), but I can't work out how to put this together.

### Environment

Environment
Application Name: Firstport
Laravel Version: 11.21.0
PHP Version: 8.3.10
Composer Version: 2.7.8
Environment: dev
Debug Mode: ENABLED
URL: localhost:8070
Maintenance Mode: OFF
Timezone: Pacific/Auckland
Locale: en

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: null
Cache: file
Database: sqlite
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 2
Sites: 1
Stache Watcher: Enabled
Static Caching: full
Version: 5.23.0 PRO

Statamic Addons
jacksleight/statamic-bard-mutator: 2.3.1
statamic-rad-pack/meilisearch: 3.3.0

**Comment by ryanmitchell** [comment, marked as answer]
If you're using the `search:results` tag then theres no way currently of passing filters to the search function - you can use conditions and query scopes on the results that come back to do filtering on the Statamic side: https://statamic.dev/conditions#overview

If you're happy to roll your own tag, you could hook into the search facade:

```php
\Statamic\Facades\Search::index('your_index')
            ->ensureExists()
            ->searchUsingApi($query, $your_mielisearch_options_as_an_array)
```

**Reply by earthwytch** [reply]
That is super helpful, thank you! 😊
